### PR TITLE
Compatibility layer to allow use of either python-sasl or pure-sasl

### PR DESCRIPTION
**Pull request by nonsleepr**

### CODE REVIEW

File: impala/sasl_compat.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+try:
+    import sasl
+
+    def build_sasl_factory(host, use_ldap, username, password, service):
```
Comment (nonsleepr) [inline impala/sasl_compat.py]: Factory for sasl_factory is probably overkill, but it allows to have all sasl-related code in one file.

**Comment by laserson** [comment]
Looks good!  There is some poor timing, but I want to merge #93 first, which will require some more changes in this PR.  Sorry about all the confusion.  Also, in case you didn't know, there is no need to close the PR to make changes.  Just feel free to push more commits to it (or force push if you rebase).  We'll just squash them at the end anyway.

**Comment by laserson** [comment]
Actually, I discussed internally, and we probably don't want to have the module fall back on pure-sasl, as it can be a liability with clients.  Let's make it so that there is a flag that you have to explicitly request the pure-sasl module if you want it.  I think we'd need to put in place some extensive testing of the pure-sasl module before we feel comfortable falling back on it.

**Comment by wesm** [comment]
I know it's been a while but any chance of rebasing this on master?

**Comment by nonsleepr** [comment]
Yes, will do that in next couple days.

**Pull request closed without merging**